File: Lexor/Lexor.Model/Responses/PayrollDashboardResponse.cs
namespace Lexor.Model.Responses
{
    /// <summary>
    /// Finance/payroll overview for the accounting dashboard, computed for the most recent PAID month.
    /// </summary>
    public class PayrollDashboardResponse
    {
        public bool HasData { get; set; }
        public int Year { get; set; }
        public int Month { get; set; }

        public decimal TotalGross { get; set; }          // sum of adjusted bruto (net + contributions + tax)
        public decimal TotalNet { get; set; }
        public decimal TotalContributions { get; set; }
        public decimal TotalTax { get; set; }
        public decimal TotalOvertime { get; set; }       // overtime pay, KM
        public decimal TotalOvertimeHours { get; set; }  // overtime worked, hours
        public int EmployeesWithOvertime { get; set; }   // how many employees had any overtime
        public decimal BurdenRate { get; set; }          // (contributions + tax) / gross, %
        public decimal AverageNet { get; set; }
        public int SlipCount { get; set; }

        public List<OvertimeLeaderItem> TopOvertime { get; set; } = new();

        public class OvertimeLeaderItem
        {
            public string FullName { get; set; } = string.Empty;
            public decimal Hours { get; set; }
            public decimal Amount { get; set; }
        }
    }
}

File: Lexor/UI/lexor_desktop/lib/models/payroll_dashboard_response.dart
class PayrollDashboardResponse {
  final bool hasData;
  final int year;
  final int month;
  final double totalGross;
  final double totalNet;
  final double totalContributions;
  final double totalTax;
  final double totalOvertime;
  final double totalOvertimeHours;
  final int employeesWithOvertime;
  final double burdenRate;
  final double averageNet;
  final int slipCount;
  final List<OvertimeLeader> topOvertime;

  PayrollDashboardResponse({
    required this.hasData,
    required this.year,
    required this.month,
    required this.totalGross,
    required this.totalNet,
    required this.totalContributions,
    required this.totalTax,
    required this.totalOvertime,
    required this.totalOvertimeHours,
    required this.employeesWithOvertime,
    required this.burdenRate,
    required this.averageNet,
    required this.slipCount,
    required this.topOvertime,
  });

  factory PayrollDashboardResponse.fromJson(Map<String, dynamic> json) {
    double d(dynamic v) => (v as num?)?.toDouble() ?? 0;
    return PayrollDashboardResponse(
      hasData: json['hasData'] as bool? ?? false,
      year: json['year'] as int? ?? 0,
      month: json['month'] as int? ?? 0,
      totalGross: d(json['totalGross']),
      totalNet: d(json['totalNet']),
      totalContributions: d(json['totalContributions']),
      totalTax: d(json['totalTax']),
      totalOvertime: d(json['totalOvertime']),
      totalOvertimeHours: d(json['totalOvertimeHours']),
      employeesWithOvertime: json['employeesWithOvertime'] as int? ?? 0,
      burdenRate: d(json['burdenRate']),
      averageNet: d(json['averageNet']),
      slipCount: json['slipCount'] as int? ?? 0,
      topOvertime: (json['topOvertime'] as List? ?? [])
          .map((e) => OvertimeLeader.fromJson(e))
          .toList(),
    );
  }
}

class OvertimeLeader {
  final String fullName;
  final double hours;
  final double amount;

  OvertimeLeader({
    required this.fullName,
    required this.hours,
    required this.amount,
  });

  factory OvertimeLeader.fromJson(Map<String, dynamic> json) => OvertimeLeader(
        fullName: json['fullName'] as String? ?? '',
        hours: (json['hours'] as num?)?.toDouble() ?? 0,
        amount: (json['amount'] as num?)?.toDouble() ?? 0,
      );
}
